src/router_xor_filter_worker.erl

```diff
@@ -85,6 +109,35 @@ init(Args) ->
 handle_call(estimate_cost, _From, State) ->
     Reply = estimate_cost(State),
     lager:info("estimating cost ~p", [Reply]),
+    {reply, Reply, State};
+handle_call(report_filter_sizes, _From, #state{filter_to_devices = FilterToDevices} = State) ->
+    Filters = get_filters(State),
+
+    Reply = [
+        {routing, enumerate_0([byte_size(F) || F <- Filters])},
+        {in_memory, [{Idx, erlang:length(Devs)} || {Idx, Devs} <- maps:to_list(FilterToDevices)]}
+    ],
+
+    {reply, Reply, State};
+handle_call(
+    {report_device_status, Device},
+    _From,
+    #state{filter_to_devices = FilterToDevices} = State
+) ->
+    BinFilters = get_filters(State),
+    DeviceEUI = deveui_appeui(Device),
+
+    Reply = [
+        {routing, [
```

Comment:
lets add the the API info (from Console) also here

Reply:
I'm fetching the Console information in the cli file, so this can only need to talk about what it's expected to have at hand. I'll think about moving it, but the reports here have the same shape because they're talking about the same thing from different places.